### ISSUE TITLE
RoPE Embeddings question

### ISSUE THREAD

**Issue by pashanitw** [lucidrains/soundstorm-pytorch#23]
In the code you provided, I observed that instead of rotating every two elements, half of the vector in the dim array is being rotated

```
def rotate_half(x):
    x1, x2 = x.chunk(2, dim=-1)
    return torch.cat((-x2, x1), dim=-1)

```
I am curious to know what is the reason behind it
